bundles/org.openhab.binding.heliosventiremote/src/main/java/org/openhab/binding/heliosventiremote/internal/HeliosVentiRemoteVariable.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    private String name;
+    private boolean writable;
+    private type datatype;
+    private byte address;
+
+    /**
+     * interpret the given byte b and return the value as string.
+     *
+     * @param b
+     * @return sting representation of byte value b in current datatype
+     */
+    public String asString(byte b) {
+        int val = b & 0xff;
+
+        if (datatype == type.Temperature) {
+            return String.format("%d °C", tempMap[val]);
+        } else if (datatype == type.Fanspeed) {
+            int i = 1;
+            while (i < fanspeedMap.length && fanspeedMap[i] < val) {
+                i++;
+            }
+            return String.format("%d", i);
+        } else if (datatype == type.BytePercent) {
+            return String.format("%d %%", val * 100 / 255);
+        } else if (datatype == type.Percent) {
+            return String.format("%d %%", val);
+        }
+
+        return "<unknown type>" + String.format("%02X ", b);
+    }
+
+    public byte getTransmitDataFor(DecimalType value) {
+        byte result = 0;
+        if (datatype == type.Temperature) {
+            int temp = (int) Math.round(value.doubleValue());
+            int i = 0;
+            while (i < tempMap.length && tempMap[i] < temp) {
+                i++;
+            }
+            result = (byte) i;
+        } else if (datatype == type.Fanspeed) {
+            int i = value.intValue();
+            if (i < 0) {
+                i = 0;
+            } else if (i > 8) {
+                i = 8;
+            }
+            result = (byte) fanspeedMap[i];
+        } else if (datatype == type.BytePercent) {
+            result = (byte) (Math.round(value.doubleValue() * 255.0 / 100));
+        } else if (datatype == type.Percent) {
+            double d = (Math.round(value.doubleValue() * 100.0));
+            if (d < 0.0) {
+                d = 0.0;
+            } else if (d > 100.0) {
+                d = 100.0;
+            }
+            result = (byte) d;
+        }
+
+        return result;
+    }
+
+    /* @return the channelUID for this variable */
+    public ChannelUID channelUID() {
+        return channelUID;
+    }
+
+    public boolean isWritable() {
+        return writable;
+    }
+
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    /**
+     *
+     * @return the name of the variable, which is also the channel name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     *
+     * @return address of the variable
+     */
+    public byte address() {
+        return address;
+    }
+
+    public DecimalType asDecimal(byte val) {
+        if (datatype == type.Temperature) {
+            return new DecimalType(tempMap[val & 0xff]);
+        } else if (datatype == type.BytePercent) {
+            return new DecimalType(val * 100.0 / 255.0);
+        } else if (datatype == type.Percent) {
+            return new DecimalType(val * 100.0);
```

Comment:
```suggestion
            return new QuantityType<>(val * 100.0, SmartHomeUnits.PERCENT);
```

Reply:
I have seen, that there is also a PercentType, so what is the difference to use PercentType or QuantityType<>(..., SmartHomeUnits.PERCENT)?

Reply:
PercentType is only used for dimmers (brightness) or rollershutters. In all other cases QuantityType is the correct way.

Reply:
Ok thanks. I tried that yesterday and I didn't manage to show the value with %-sign except.when I.changed the channeltype to a Dimmer. Can you point me to an example where % is used properly?